addons/account/models/account_journal_dashboard.py

```diff
@@ -427,7 +432,7 @@ def _fill_sale_purchase_dashboard_data(self, dashboard_data):
         to_check_vals = {
             vals['journal_id'][0]: vals
             for vals in self.env['account.move'].read_group(
-                domain=[('journal_id', 'in', sale_purchase_journals.ids), ('to_check', '=', True)],
+                domain=[('journal_id', 'in', sale_purchase_journals.ids), ('company_id', '=', self.env.companies.ids), ('to_check', '=', True)],
```

Comment:
Most important part is the `in` instead of `=` ofc
```suggestion
                domain=[
                    ('journal_id', 'in', sale_purchase_journals.ids), 
                    ('company_id', 'in', self.env.companies.ids), 
                    ('to_check', '=', True)
                ],
```

Reply:
:man_facepalming: